include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aListener  A pointer to a TCP listener structure.
+ * @param[in]  aArgs      A pointer to a structure of arguments.
+ * @param[in]  aContext   A pointer to application-specific context.
+ *
+ * @retval OT_ERROR_NONE    Successfully opened the TCP listener.
+ * @retval OT_ERROR_FAILED  Failed to open the TCP listener.
+ */
+otError otTcpListenerInitialize(otInstance *aInstance, otTcpListener *aListener, otTcpListenerInitializeArgs *aArgs);
+
+/**
+ * Sets the "accept ready" callback on a TCP listener.
+ *
+ * @param[in]  aInstance             A pointer to the OpenThread instance to which the TCP listener belongs.
+ * @param[in]  aListener             A pointer to the TCP listener structure whose callback to set.
+ * @param[in]  aAcceptReadyCallback  The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetAcceptReadyCallback(otInstance *aInstance, otTcpListener *aListener, otTcpAcceptReady aAcceptReadyCallback);
+
+/**
+ * Sets the "accept done" callback on a TCP listener.
+ *
+ * @param[in]  aInstance            A pointer to the OpenThread instance to which the TCP listener belongs.
+ * @param[in]  aListener            A pointer to the TCP listener structure whose callback to set.
+ * @param[in]  aAcceptDoneCallback  The new callback.
+ *
+ * @retval OT_ERROR_NONE    Successfully set the callback.
+ * @retval OT_ERROR_FAILED  Failed to set the callback.
+ */
+otError otTcpSetAcceptDoneCallback(otInstance *aInstance, otTcpListener *aListener, otTcpAcceptDone aAcceptDoneCallback);
+
+/**
+ * Causes incoming TCP connections that match the specified IP address and port
+ * to trigger this TCP listener's callbacks.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP listener belongs.
+ * @param[in]  aListener  A pointer to the TCP listener structure that should begin listening.
+ * @param[in]  aSockName  The address and port on which to listen for incoming connections.
+ *
+ * @retval OT_ERROR_NONE    Successfully initiated listening on the TCP listener.
+ * @retval OT_ERROR_FAILED  Failed to initiate listening on the TCP listener.
+ */
+otError otTcpListen(otInstance *aInstance, otTcpListener *aListener, const otSockAddr *aSockName);
+
+/**
+ * Causes this TCP listener to stop listening for incoming connections.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP listener belongs.
+ * @param[in]  aListener  A pointer to the TCP listener structure that should stop listening.
+ *
+ * @retval OT_ERROR_NONE    Successfully stopped listening on the TCP listener.
+ * @retval OT_ERROR_FAILED  Failed to stop listening on the TCP listener.
+ */
+otError otTcpStopListening(otInstance *aInstance, otTcpListener *aListener);
+
+/**
+ * Deinitializes this TCP listener. This means that OpenThread no longer keeps
```

Comment:
nit: have brief in separate block above

Reply:
Done. I'm not 100% what you meant by "in separate block", can you please check that what I've done is acceptable?